Firestore/Source/Model/FSTFieldValue.mm

```diff
@@ -1027,4 +959,114 @@ - (NSComparisonResult)compare:(FSTFieldValue *)other {
 
 @end
 
+@interface FSTDelegateValue ()
+@property(nonatomic, assign, readonly) FieldValue internalValue;
+@end
+
+@implementation FSTDelegateValue
++ (instancetype)delegateValue:(FieldValue &&)value {
+  return [[FSTDelegateValue alloc] initWithValue:std::move(value)];
+}
+
+- (id)initWithValue:(FieldValue &&)value {
+  self = [super init];
+  if (self) {
+    _internalValue = std::move(value);
+  }
+  return self;
+}
+
+- (FieldValue::Type)type {
+  return self.internalValue.type();
+}
+
+- (FSTTypeOrder)typeOrder {
+  switch (self.internalValue.type()) {
+    case FieldValue::Type::Null:
+      return FSTTypeOrderNull;
+    case FieldValue::Type::Boolean:
+      return FSTTypeOrderBoolean;
+    case FieldValue::Type::Integer:
+    case FieldValue::Type::Double:
+      return FSTTypeOrderNumber;
+    case FieldValue::Type::Timestamp:
+    case FieldValue::Type::ServerTimestamp:
+      return FSTTypeOrderTimestamp;
+    case FieldValue::Type::String:
+      return FSTTypeOrderString;
+    case FieldValue::Type::Blob:
+      return FSTTypeOrderBlob;
+    case FieldValue::Type::Reference:
+      return FSTTypeOrderReference;
+    case FieldValue::Type::GeoPoint:
+      return FSTTypeOrderGeoPoint;
+    case FieldValue::Type::Array:
+      return FSTTypeOrderArray;
+    case FieldValue::Type::Object:
+      return FSTTypeOrderObject;
+  }
+  UNREACHABLE();
+}
+
+- (BOOL)isEqual:(id)other {
+  // Simplification: We'll assume that (eg) FSTBooleanValue(true) !=
```

Comment:
Repeating this comment throughout isn't that useful, especially when it contributes a reference to a type we want deleted.

I think it's rational to note that the swap to FSTDelegateValue and removal of FSTBooleanValue has to happen in one PR in the PR comments proper, or in a warning at the top of this file (like in notes to porters) but nobody actually needs to look at the implementation of isEqual to do the next type, so this isn't directly helping.

Ditto below in `compare:`.